File: problem solving/datastructures/strings/isPalindrome.js
/**
 * problem: https://leetcode.com/problems/palindrome-number
 * concepts: string, number to string casting
 * runtime: 42.38% memory: 74.59%
 */
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function(x) {
    let string_x = x+"";
    let reverse_x = string_x.split('').reverse().join("");
    return string_x === reverse_x;
};

File: problem solving/algorithms/two-pointers/isPalindrome.py
'''
    problem: https://leetcode.com/problems/valid-palindrome/
    concepts: two pointers, string
    performance: 59.25% runtime, 86.18% memory
'''
class Solution:
    def isPalindrome(self, phrase: str) -> bool:
        s = ""
        for char in phrase:
            if char.isalnum():
                s += char
        s = s.lower()
        i = 0
        n = len(s)
        l = i
        r = n-1-i
        while l <= r:
            if s[l] != s[r]:
                return False
            i += 1
            l = i
            r = n-1-i
        return True